array_api_strict/tests/test_array_object.py

```diff
@@ -101,6 +101,29 @@ def test_validate_index():
     assert_raises(IndexError, lambda: a[idx])
 
 
+@pytest.mark.parametrize("device", [None, "CPU_DEVICE", "device1", "device2"])
+@pytest.mark.parametrize(
+    "integer_index",
+    [
+        1,
+        np.int8(0),
+        np.uint8(0),
+        np.int16(0),
+        np.uint16(0),
+        np.int32(0),
+        np.uint32(0),
+        np.int64(0),
+        np.uint64(0),
+    ],
+)
+def test_indexing_ints(integer_index, device):
+    # Ensure indexing with different integer types works on all Devices.
+    device = None if device is None else Device(device)
+
+    a = arange(5, device=device)
+    a[integer_index]
```

Comment:
Let's try to be airtight, esp given that the original issue was reported to occur for `a[(integer_index,)]`:

```suggestion
    assert a[integer_index] == a[0]
    assert a[(integer_index,)] == a[0]
```

Reply:
:facepalm: of course!